include/aarch32_gcc_accessor_macros.h

```diff
@@ -80,7 +80,8 @@
     uint32_t address = addr;                                                   \
     uint32_t val = 0;                                                          \
     __asm__ __volatile__(                                                      \
-        "ldr %[val], %[address]\n"                                             \
+        "mov r0, %[address]\n"                                                 \
+        "ldr %[val], [r0]\n"                                                   \
```

Comment:
I based my assembly off of this tutorial: [Link](https://azeria-labs.com/memory-instructions-load-and-store-part-4/). They load the address into a register, and then access it by referencing the location in the register. I tried writing it like you put above, but I wasn't convinced it was storing everything correctly. Let me dig a little more and see if I can test it.

Reply:
Written PR's long way:
```
   c:	d2818c00 	mov	x0, #0xc60                 	// #3168
  10:	aa0003e0 	mov	x0, x0
  14:	f9400001 	ldr	x1, [x0]
```

Written your abbreviated way:
```
   c:	d2818c00 	mov	x0, #0xc60                 	// #3168
  10:	f9400001 	ldr	x1, [x0]
```

Long story short, you are right. It seems that the compiler automatically generates that mov to a register. My way has a redundant move and is unnecessary. I think originally I was generating the ASM with `%[address]` instead of `[%[address]]` which was throwing me off. I'll correct this in my PR.

Reply:
👍